### PR TITLE
[Incubator][VC] Clean up - Refactor resource controller - Part 1

### PR DESCRIPTION
**Pull request by Fei-Guo**
This change mainly does the followings:

1) Move downward syncer logic to a separate dws.go for better code structure.
2) Move WaitForCacheSync to the right sync thread.
3) Add more comments.

This change touches configmap/events/namespace/event resources. There are no functional change.

### CODE REVIEW

File: incubator/virtualcluster/pkg/syncer/resources/configmap/controller.go [hunk 1]
```diff
@@ -60,108 +53,18 @@ func Register(
 		return
 	}
 	c.multiClusterConfigMapController = multiClusterConfigMapController
-
 	c.configMapLister = configMapInformer.Lister()
-	c.configMapSynced = configMapInformer.Informer().HasSynced
```
Comment (zhuangqh) [inline incubator/virtualcluster/pkg/syncer/resources/configmap/controller.go]: we still need to wait for configmapSynced, or the configMapLister cannot work.

Comment (Fei-Guo) [reply incubator/virtualcluster/pkg/syncer/resources/configmap/controller.go]: Fixed

**Pull request merged**